design/one-pager-stack-status-spec.md

```diff
@@ -0,0 +1,114 @@
+# Stack UI Schema Status Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#929](https://github.com/crossplaneio/crossplane/issues/929)
+
+## Problem
+
+For a given CRD I want to be able to expose various statuses to a frontend GUI. Initially it makes
+sense to at least have the Ready state available. In addition to the value of the Ready state it
+would also be helpful to have some additional details about that state if applicable.
+
+This also assumes that statuses are following the podCondition object convention as described here:
+[podcondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#podcondition-v1-core)
+
+#### For example:
+
+If a CRD has the condition type 'Ready' defined but has a status of that type of
+_"False"_, I could show some visual indication to a user through a GUI that the CRD is **NOT Ready**.
+
+When that status changes to _"True"_ then I can adjust that visual cue to now reflect that the CRD is
+in fact **Ready**.
+
+As of right now there is no defined spec to be able to drive status information for a given stack or
+resource from a GUI perspective. It would be useful to add keys to the existing `ui-schema.yaml`
+file that can be used for this purpose.
+
+We can leverage the types from the array of **conditions** within the **conditionedStatus** key
+defined in **status** as long as we follow the crossplane specific spec.
+
+For more arbitrary stacks or stacks that don't contain the conditionedStatus key, you can also pass
+the full JSONPath as a string within that array as long as it follows the podcondition convention.
+
+This 1-pager will identify the keys that should be added and what that structure looks like.
+
+## Design
+
+#### Add two new keys to the `ui-schema.yaml` file:
+
+- `resourceStatusTypes` _(this is an array of condition types **OR** JSONPaths)_
+- `stackStatusTypes` _(this is an array of condition types **OR** JSONPaths)_
+
+#### The arrays could potentially look like this inside of the `ui-schema.yaml` file:
+
+```yaml
+version: 0.4
+configSections: 
+- title: Title
+  description: Description
+resourceStatusTypes:
+- Ready
+stackStatusTypes:
+- .status.conditionedStatus.conditions[?(@.type=='Ready')]
```

Comment:
what do you think about the usability of overloading these new `*Types` keys with both type names and query paths?  Do you think it would be more intuitive and understandable to have 4 new keys instead?

* `resourceConditionTypes`
* `resourceConditionPaths`
* `stackConditionTypes`
* `stackConditionPaths`

Reply:
I am going to be revising what these arrays will contain to eliminate the confusion here